new-frontiers-14/frontier-station-14
# BankSystem: cached balances

## About the PR



Adds caching to the server's BankAccountComponent handler for ComponentGetState.

Logic should largely follow previous behaviour - only the user controlling a given entity can update its value in the database/cache.  Adds a dictionary in the BankSystem (yes, likely poor form, may need a refactor) to prevent unnecessary DB lookups when the last requested value matches the most recently written one (even if the value hasn't gone through into the DB)

## Why / Balance


Updating BankAccounts was a pain, and the ComponentGetState function seems to get called not only on writing values but on queries as well.

Should clear up any issues for #1399 - will behave a lot better with many clients.

## How to test


1. Deposit money with a character, attach a breakpoint on Content.Server/_NF/Bank/BankSystem.cs:78 or so to ensure that only one DB hit results from the deposit.
2. Disconnect, close the server, reopen it, check your character's money, should be consistent with the deposited value.

## Media



## Breaking changes


**Changelog**



Largely an internal feature, no changelog needed.
